### ISSUE TITLE
[1.18.2] Working Reap Tag for Harvestcraft2 

### ISSUE THREAD

**Issue by KuwuroUsagi** [henkelmax/reap#15]
It doesn't seem as though #minecraft:crops or #forge:crops are working, has anyone found a working tag so that I and others don't have to add hundreds of blocks to config?
For example: #mysticalagriculture:crops is working fine for Mystical Agriculture
All HC2 crops are tagged with forge:crops so it's a tad saddening it isn't working with reap out of the box

**Comment by henkelmax** [comment]
So whats your actual question?
Does it work with `#forge:crops`  or not?

**Comment by KuwuroUsagi** [comment]
it didn't seem to be working with either minecraft or forge crops tags despite hc2 being tagged with both

**Comment by henkelmax** [comment]
Can you send me your config?

**Comment by KuwuroUsagi** [comment]
```

[crop_harvesting]
	#The blocks that should get harvested by right-clicking
	whitelist = ["#minecraft:crops", "#mysticalagriculture:crops", "#forge:crops", "#pamhc2crops:"]
	#If the held tool should be considered when right-click harvesting
	consider_tool = true

[tree_harvesting]
	#Examples: 'minecraft:oak_log', '#minecraft:logs'
	log_types = ["minecraft:acacia_log", "minecraft:birch_log", "minecraft:dark_oak_log", "minecraft:jungle_log", "minecraft:oak_log", "minecraft:spruce_log", "minecraft:crimson_stem", "minecraft:warped_stem"]
	#Examples: 'minecraft:dirt', '#forge:sand/colorless'
	ground_types = ["minecraft:dirt", "minecraft:grass_block", "minecraft:coarse_dirt", "minecraft:podzol", "minecraft:mycelium", "minecraft:warped_nylium", "minecraft:crimson_nylium", "minecraft:netherrack"]
	#The tools which the player is allowed to harvest trees
	allowed_tree_tools = ["minecraft:wooden_axe", "minecraft:golden_axe", "minecraft:stone_axe", "minecraft:iron_axe", "minecraft:diamond_axe", "minecraft:netherite_axe"]
	#If the tree harvester should be enabled
	enabled = false
	#The maximum amount of logs one harvest is allowed to do
	#Range: 0 ~ 1024
	max_harvesting_count = 128

	[tree_harvesting.dynamic_breaking_speed]
		#If bigger trees should be harder to break
		enabled = true
		#The maximum amount of time a tree should take to harvest
		#Range: 1.0 ~ 100.0
		min_speed = 10.0
		#The amount of breaking time that gets added per harvested log
		#Range: 0.0 ~ 100.0
		per_log = 0.1


```

I'll be updating my server/modpack with a few mods and updates to existing ones, so perhaps I can test again in the future, but for now I'm stepping away from it for a little while, I also looked into pam's side as well, but didn't have a ton of luck. The current "#pamhc2crops:" in cfg is just my most recent effort at trying to get some variant working, nothing I tried worked so I never updated it to anything more sensible.

**Comment by henkelmax** [comment]
> `whitelist = ["#minecraft:crops", "#mysticalagriculture:crops", "#forge:crops", "#pamhc2crops:"]`

`#pamhc2crops:` is invalid. That might be the issue.

**Comment by lkjfdsaofmc** [comment]
Coincidentally I was just running into this issue on a custom modpack my friends and I are running.  I've spent the past few hours working on a fix and I came to the discovery that in this version Pam's HarvestCraft 2 Crops simply do not have any tags.  The only tag each crop has is one for that single crop (E.I. "pamshc2crops:pamblueberrycrop").  I used an excel spreadsheet algorithm with a manually typed list of the crops in the game to generate a line of text which contains all the tags for each individual crop.  If you copy paste the line from this text file into the whitelist list for the reap mod it will allow it to work with every Pam's crop.

[PamsTags.txt](https://github.com/henkelmax/reap/files/9485523/PamsTags.txt)
